src/ds/d500/d500-device.cpp

```diff
@@ -409,7 +409,7 @@ namespace librealsense
             _hw_monitor = std::make_shared<hw_monitor_extended_buffers>(
                 std::make_shared<locked_transfer>(
                     std::make_shared<command_transfer_over_xu>(
-                        raw_sensor, depth_xu, DS5_HWMONITOR),
+                        *raw_sensor, depth_xu, DS5_HWMONITOR),
```

Comment:
Why not shared_ptr like d400
![image](https://github.com/IntelRealSense/librealsense/assets/64067618/7ab130d6-3fab-451c-880f-1edf42e5af8e)

Reply:
This is like the D400 (only different indentation, will fix).
The `locked_transfer` gets a `shared_ptr`. The `command_transfer_over_xu` still gets a reference. Did not change because I am not familiar with the purpose of each class and over all OSs. I did not want to do all in on PR.

Reply:
OK let's leave for now and if we see issues we will get back to it